### PR TITLE
add changes in files for avro-1.7.7 to use latest PPC hadoop packages

### PR DESCRIPTION
**Pull request by GauravPande**
Changes made are : 
1. build.sh 
- added forrest.home
  1. avro/lang/java/pom.xml 
  2. changed versions for hadoop1 (2.6.0), hadoop2 (2.6.0), protobuf (2.5.0), snappy-java (1.1.1.3)
  3. Other changes are as per the commit in branch 1.7.4-IBM
     https://github.com/ibmsoe/avro/commit/2b897f0020fbec49b1ac150a0bd1bfe00e6b07ef
  4. lang/java/protobuf/src/test/java/org/apache/avro/protobuf/Test.java
  5. changes made to make it compatible with protoc - 2.5.0

### CODE REVIEW

File: lang/java/pom.xml [hunk 1]
```diff
@@ -42,8 +42,8 @@
       To build the avro-mapred module against Hadoop 2 specify
       -Dhadoop.version=2 or leave unspecified to build against Hadoop 1
     -->
-    <hadoop1.version>0.20.205.0</hadoop1.version>
```
Comment (pascal-oliva) [inline lang/java/pom.xml:45]: Why did you change the hadoop1.version to 2.6.0 ?

Comment (pascal-oliva) [inline lang/java/pom.xml:45]: there is hadoop2.version tag to set with 2.6.0

Comment (GauravPande) [inline lang/java/pom.xml:45]: Yes, I should just change hadoop2.version to 2.6.0, leaving the hadoop1.version as it is (that is, 2.0.1-alpha)
I will make these changes and commit it.

File: lang/java/tools/pom.xml [hunk 2]
```diff
@@ -145,7 +145,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-core</artifactId>
```
Comment (pascal-oliva) [inline lang/java/tools/pom.xml:148]: Are you sure that hadoop-core is not mandatory for a specific case ? 
if it is used in some cases ; we have just to add hadoop-client as dependency (not replace hadoop-core).

Comment (GauravPande) [inline lang/java/tools/pom.xml:148]: I guess hadoop-core is not required for hadoop2 profile, it is required for hadoop1, and we are building avro-1.7.7 for hadoop2.

I was following below commit url to make the changes 
https://github.com/ibmsoe/avro/commit/2b897f0020fbec49b1ac150a0bd1bfe00e6b07ef

But to be on safer side I will add both hadoop-core and hadoop-client as the dependencies.

Comment (GauravPande) [inline lang/java/tools/pom.xml:148]: Adding hadoop-core doesnt seems to be working, it tries to find the jar which is not available. And for the given hadoop version (2.6.0) hadoop-client works fine.

Comment (pascal-oliva) [inline lang/java/tools/pom.xml:148]: you can use the dependency into the specific profile : see ./lang/java/mapred/pom.xml
the use of hadoop-core and hadoop-client regarding the profile hadoop1 or hadoop2

Comment (GauravPande) [inline lang/java/tools/pom.xml:148]: As per your suggestion, I have added profiles and dependencies
